src/main/java/Problem1/MinStack.java

```diff
@@ -4,28 +4,44 @@
     // okay to add member variables
     // can only use Stack interface and ArrayStack from this folder
     // do not use Java Stack
+    private int min = 999999999;
 
     public MinStack(int size) {
-        // homework
-        super(size); // place holder
+        super(size);
     }
 
     @Override
     public boolean push(Integer val) {
-        // homework
-        return false; // place holder
+        if(val < min){
+            min = val;
+        }
+        super.push(val);
+        return true;
```

Comment:
I'm a little suspicious of this algorithm. Will circle back to try to do a bit more tests.